app/main/views/suppliers.py

```diff
@@ -276,25 +277,78 @@ def edit_supplier_organisation_size():
     form = CompanyOrganisationSizeForm()
 
     if request.method == 'POST':
+        api_error = None
         if form.validate_on_submit():
-            data_api_client.update_supplier(supplier_id=current_user.supplier_id,
-                                            supplier={"organisationSize": form.organisation_size.data},
-                                            user=current_user.email_address)
+            try:
+                data_api_client.update_supplier(supplier_id=current_user.supplier_id,
+                                                supplier={"organisationSize": form.organisation_size.data},
+                                                user=current_user.email_address)
+
+            except APIError as e:
```

Comment:
I think we agreed to do do this for API failures?
```
except APIError as e:
    abort(e.status_code)
```

Reply:
To be consistent with elsewhere

Reply:
Actually, this might not be part of your ticket and will go away after a rebase...?